docs/cumulusci_flow.rst

```diff
@@ -0,0 +1,273 @@
+
+.. image:: images/salesforce-org-process.png
+   :scale: 50 %
+   :alt: Salesforce.org Dev/Release Process Diagram
+
+
+
+Project Considerations
+----------------------
+CumulusCI Flow was designed for use with Salesforce development projects, which inject some unique considerations into finding the right branching model:
+
+* You cannot re-cut a Salesforce managed package release with the same version as a prior release.  As a result, Git Tags are the best representation of our releases in a repository since they are a read only reference to the exact code we put into a given release.
+* Most Salesforce managed package projects use Push Upgrades to keep all users of the package on the latest release.  This eliminates the need to consider supporting multiple past versions.
+* Releasing managed packages has some overhead involved including manual checks by release managers to ensure nothing gets permanently locked into the package in a release.  As a result, true continuous delivery isn't an option.  Whether you're on a team that wants to deliver quickly (e.g. a two week sprint cycle) or at team that makes several larger releases a year, CumulusCI offers functionality to help cut releases for all products with any changes.
+
+
+
+Main Builds
+-----------
+The main goal of the CumulusCI Flow is to always have the main branch ready to cut into a package.
+This way, we can merge a fix and cut an emergency release at any time in the development process.
+
+To test that we can package main, we upload a beta release on every commit to main and then test that beta release in a variety of Salesforce org environments concurrently.
+This build ranges from 15 minutes to 2 hours depending on the project and a passing build is proof we can package main at any point in time.
+
+When the upload of the beta release is completed, the main branch is auto-merged_ into all open feature branches.
+New betas are published on Github as a Github Release along with automatically generated release notes created by parsing the body of all Pull Requests merged since the last production release
+
+
+
+CumulusCI and Main Builds
+^^^^^^^^^^^^^^^^^^^^^^^^^
+There are four main flows that facilitate main builds:
+
+* ``ci_master``: Deploys the main branch and all dependencies into the packaging org including incrementally deleting any metadata deleted in the commit.  The end result is a package that is ready to be uploaded from the packaging org.
+* ``release_beta``: Uploads a beta release of the code staged in the packaging org, creates a Github Tag and Release, generates release notes, adds them to the release, and merges main into all feature branches.
+* ``ci_beta``: Installs the beta and all dependencies into a fresh scratch org and runs the Apex tests.
+* ``ci_beta_install``: Installs the beta and all dependencies into a fresh scratch org. This is used to prepare environments for non-Apex testing such as automated browser tests.
```

Comment:
This flow doesn't exist. `install_beta`?

Reply:
Just take this one out; it's not particularly related to CI